### DISCUSSION TITLE
Replace a ZBDongle-E sonoff dongle with another ZBDongle-E (EZSP)

### DISCUSSION THREAD

**Discussion by Ferywir** [Koenkk/zigbee2mqtt#18079] [Devices]
Hello everyone,

I would like to know if there is a procedure to replace a ZBDongle-E (sonoff EZSP) dongle with the same one. (in case of breakage of the dongle).

I read in the FAQ that if we change the coordinator to the same coordinator, there is no need to re-pair, however, the IEEE address of the coordinator must be changed. For a CC2652, CC1352, CC2538 coordinator there is a software available however for the EZSP what should be done?

https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

Thanks for your help
Ferywir

**Comment by Ferywir** [comment]
Good morning,

Has anyone ever had to change their EZSP dongle? Whether it's a POP key or Sonoff ZBDongle-E?

THANKS

**Comment by Hedda** [comment]
> would like to know if there is a procedure to replace a ZBDongle-E (sonoff EZSP) dongle with the same one. (in case of breakage of the dongle).
> 
> I read in the FAQ that if we change the coordinator to the same coordinator, there is no need to re-pair,

See existing discussion for migration workaround to backup and restore ezsp adapter with third-party command-line tool -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478

**Comment by Bekie74** [comment]
Hi @Ferywir 
Did you manage to change the IEEE address on your Dongle-E?  
If so, I would be interested in knowing exactly how you did it.
The internet is loaded with tutorials on how to change the IEEE on a Dongle-P, but on a Dongle-E?  Can't find anything.
Any help would be greatly appreciated.
thx!

**Reply by Hedda** [reply]
NabuCasa’s universal-silabs-flasher can supposedly do it:

- https://github.com/NabuCasa/universal-silabs-flasher

at least acordingly to their readme:

- https://github.com/NabuCasa/universal-silabs-flasher/blob/dev/README.md#writing-ieee-address

Probably best to ask @Nerivec if related to Zigbee2MQTT as he is working a fork for other things.

- https://github.com/Nerivec/silabs-firmware-builder/issues

**Reply by Nerivec** [reply]
@Bekie74 You can write it with the above, or with [Ember ZLI](https://github.com/Nerivec/ember-zli/wiki#installation), see the docs here: https://github.com/Nerivec/ember-zli/wiki/Stack#write-eui64-nvm3-token
_Make sure you have a firmware 7.4.x or above for Ember ZLI._
_Make sure you use the format whichever tool you use is expecting._
